dsp/window/window_complex_test.go

```diff
@@ -0,0 +1,150 @@
+		&testingCaseCmplx{Name: "NuttallComplex", Function: NuttallComplex,
+			ExpectedResult: []float64{0.000000, 0.001706, 0.011614, 0.043682, 0.117808, 0.250658, 0.441946, 0.664015, 0.864348, 0.984019,
+				0.984019, 0.864348, 0.664015, 0.441946, 0.250658, 0.117808, 0.043682, 0.011614, 0.001706, 0.000000}},
+		&testingCaseCmplx{Name: "BlackmanNuttallComplex", Function: BlackmanNuttallComplex,
+			ExpectedResult: []float64{0.000363, 0.002885, 0.015360, 0.051652, 0.130567, 0.266629, 0.457501, 0.675215, 0.869392, 0.984644,
+				0.984644, 0.869392, 0.675215, 0.457501, 0.266629, 0.130567, 0.051652, 0.015360, 0.002885, 0.000363}},
+		&testingCaseCmplx{Name: "FlatTopComplex", Function: FlatTopComplex,
+			ExpectedResult: []float64{0.004000, -0.011796, -0.078650, -0.212762, -0.328021, -0.178010, 0.531959, 1.862876, 3.422134, 4.490270,
+				4.490270, 3.422134, 1.862876, 0.531959, -0.178010, -0.328021, -0.212762, -0.078650, -0.011796, 0.004000}},
+	}
+
+	//testing table (gauss window)
+	var ttg []*testingCaseCmplxGauss = []*testingCaseCmplxGauss{
+		&testingCaseCmplxGauss{Name: "GaussComplex (sigma=0.3)", Sigma: 0.3,
+			ExpectedResult: []float64{0.003866, 0.011708, 0.031348, 0.074214, 0.155344, 0.287499, 0.470444, 0.680632, 0.870660, 0.984728,
+				0.984728, 0.870660, 0.680632, 0.470444, 0.287499, 0.155344, 0.074214, 0.031348, 0.011708, 0.003866}},
+		&testingCaseCmplxGauss{Name: "GaussComplex (sigma=0.5)", Sigma: 0.5,
+			ExpectedResult: []float64{0.135335, 0.201673, 0.287499, 0.392081, 0.511524, 0.638423, 0.762260, 0.870660, 0.951361, 0.994475,
+				0.994475, 0.951361, 0.870660, 0.762260, 0.638423, 0.511524, 0.392081, 0.287499, 0.201673, 0.135335}},
+		&testingCaseCmplxGauss{Name: "GaussComplex (sigma=1.2)", Sigma: 1.2,
+			ExpectedResult: []float64{0.706648, 0.757319, 0.805403, 0.849974, 0.890135, 0.925049, 0.953963, 0.976241, 0.991381, 0.999039,
+				0.999039, 0.991381, 0.976241, 0.953963, 0.925049, 0.890135, 0.849974, 0.805403, 0.757319, 0.706648}},
+	}
+
+	// run tests on testing tables
+	for _, tc := range tt {
+		t.Run(tc.Name, func(t *testing.T) {
+			//copy src, because window functions change data inplace
+			srcCpy := append([]complex128(nil), src...)
+			dst := tc.Function(srcCpy)
+
+			//check that function returned the slice with the same array, not a newly allocated
+			addressSrc := unsafe.Pointer(&srcCpy)
+			addArraySrc := unsafe.Pointer(*(*uintptr)(addressSrc))
+			addressDst := unsafe.Pointer(&dst)
+			addArrayDst := unsafe.Pointer(*(*uintptr)(addressDst))
+			if addArraySrc != addArrayDst {
+				t.Errorf("unexpected result for window function %q : function returned a slice with new underling array at %v instead of source array at %v modification ",
+					tc.Name, addArrayDst, addArraySrc)
+			}
+
+			//check the data
+			if !equalApprox(dst, tc.ExpectedResult, tol) {
+				t.Errorf("unexpected result for window function %q: result \n%f\n is different from expectation \n%f\n more then it is allowed by test tolerance %e",
```

Comment:
`t.Errorf("unexpected result for window function %q:\ngot:%v\nwant:%v", tc.name, dst, test.want)`